HW_5/index.js

```diff
@@ -0,0 +1,159 @@
+"use strict";
+
+const form = document.querySelector('.create-task-form');
+const taskList = document.querySelector('.collection');
+const clearBtn = document.querySelector('.clear-tasks');
+const filter = document.querySelector('.filter.input');
+const taskInput = document.querySelector('.task-input');
+
+document.addEventListener('DOMContentLoaded', renderTasks);
+form.addEventListener('submit', createTask);
+clearBtn.addEventListener('click', clearAllTasks);
+//taskList.addEventListener('click', clearSingleTask);
+
+function getTasksFromLocalStorage() {
+    return localStorage.getItem('tasks') !== null ? JSON.parse(localStorage.getItem('tasks')) : [];
+}
+
+function setTasksToLocalStorage(tasks) {
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+function createSingleTaskElement(task) {
+    const li = document.createElement('li');
+    li.className = 'collection-item';
+
+    const textElement = document.createElement('span');
+    textElement.className = 'text-item';
+    textElement.textContent = task;
+    li.appendChild(textElement);
+
+    const editElement = document.createElement('span');
+    editElement.className = 'edit-item';
+    editElement.innerHTML = '<i class="fa fa-edit"></i>';
+    editElement.addEventListener('click', () => editOneElement(li, textElement, editElement));
+
+    li.appendChild(editElement);
+
+    const deleteElement = document.createElement('span');
+    deleteElement.className = 'delete-item';
+    deleteElement.innerHTML = '<i class="f fa-remove"></i>';
+    li.appendChild(deleteElement);
+    deleteElement.addEventListener('click', clearSingleTask);
+
+    taskList.appendChild(li);
+
+}
+
+function editOneElement(edit, textElement, editButton) {
+    const editTask = document.createElement('input');
+    textElement.remove();
+    editButton.remove();
+    editTask.setAttribute('type', 'text');
+    editTask.value = textElement.textContent;
+    editTask.addEventListener("keypress", function (event) {
+        if (event.key === "Enter") {
+            event.preventDefault();
+            editTask.remove();
+            console.log(edit.textContent)
+            textElement.textContent = editTask.value;
+            edit.insertBefore(editButton, edit.children[0]);
+            edit.insertBefore(textElement, edit.children[0]);
+
+            const nodes = Array.prototype.slice.call(taskList.children);
+            const index = nodes.indexOf(textElement.parentElement);
+            updateTaskInLocalStorage(textElement.textContent, index);
+        }
+    })
+
+    edit.insertBefore(editTask, edit.children[0]);
+    console.log(edit)
+
+}
+
+
+function renderTasks() {
+    let tasks;
+
+    if (localStorage.getItem('tasks') !== null) {
+        tasks = JSON.parse(localStorage.getItem('tasks'));
+    } else {
+        tasks = [];
+    }
```

Comment:
```suggestion
const tasks = getTasksFromLocalStorage();
```

Функція getTasksFromLocalStorage ма/
є в собі цей же функціонал